### ISSUE TITLE
Multiple attachment messages point back to a single attachment

### ISSUE THREAD

**Issue by tnvu** [BlueBubblesApp/bluebubbles-app#2507]
When multiple images are sent together (or one after another, I can't tell)... the server seems to get confused and will assign multiple images to the same attachment. for example, in iMessage, I see a series of photos correctly (image1.jpg, image2.jpg, image3.jpg, image4.jpg) but when I try to access them via the android client or the windows client... I get four images, but they all access a single image (for example, all would be image1.jpg). I can confirm this on the server side logs because I see the request for the same attachment when I try to re-download attachment from the server

Looking at the message info, the info is duplicated.

**Comment by tnvu** [comment]
it looks like its a client problem... i examined the Message entity and it correctly lists two attachments, but doesn't seem to display the right one in the UI... so when I select it to re-download, it will obviously get the item displayed multiple times

**Comment by tnvu** [comment]
Nevermind, it definitely looks like a server side issue now.

I didn't correctly notice that the attachment guids are different by a single character and so the client is definitely requesting the correct guids, but the server is returning the wrong file.

I looked in ~/Library/Application Support/bluebubbles-server/Convert and can see the attachment images there.

When I open the images, it correctly displays two different images, but that's not what the server is serving up.

**Comment by tnvu** [comment]
I think I found the issue --

MessageRepository.getAttachment() accepts an attachment GUID but discards the attachment prefix to perform a lookup based on the "actual" guid. It then queries the database and then calls query.getOne() which return the same attachment every single time.

**Comment by zlshames** [comment]
> I think I found the issue --
> 
> MessageRepository.getAttachment() accepts an attachment GUID but discards the attachment prefix to perform a lookup based on the "actual" guid. It then queries the database and then calls query.getOne() which return the same attachment every single time.

This is code on the server, so it is a server bug?

**Comment by tnvu** [comment]
Yeah, I think so. I initially found the behavior on the android client and wasn't sure what was going on. I brought up the issue in both #android-support and #server-support and was told to file a client bug.

I eventually dug into the code to see what was going on and found the problem. Not sure how to change it to a server bug.
